crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1245,27 +1241,82 @@ impl Segment2d {
     ///
     /// Panics if `point1 == point2`
     #[inline(always)]
+    #[deprecated(since = "0.15.1", note = "Use the `new` constructor instead")]
     pub fn from_points(point1: Vec2, point2: Vec2) -> (Self, Vec2) {
-        let diff = point2 - point1;
-        let length = diff.length();
+        (Self::new(point1, point2), (point1 + point2) / 2.)
+    }
 
-        (
-            // We are dividing by the length here, so the vector is normalized.
-            Self::new(Dir2::new_unchecked(diff / length), length),
-            (point1 + point2) / 2.,
-        )
+    /// Create a new `Segment2d` at the origin from a `direction` and `length`
+    #[inline(always)]
+    pub fn from_direction_and_length(direction: Dir2, length: f32) -> Segment2d {
+        let half_length = length / 2.;
+        Self::new(direction * -half_length, direction * half_length)
     }
 
     /// Get the position of the first point on the line segment
     #[inline(always)]
     pub fn point1(&self) -> Vec2 {
-        *self.direction * -self.half_length
+        self.point1
     }
 
     /// Get the position of the second point on the line segment
     #[inline(always)]
     pub fn point2(&self) -> Vec2 {
-        *self.direction * self.half_length
+        self.point2
+    }
+
+    /// Get the segment's center
+    #[inline(always)]
+    pub fn center(&self) -> Vec2 {
+        (self.point1() + self.point2()) / 2.
+    }
```

Comment:
I would add a `midpoint` alias since Glam has it for vectors. E.g. this could be `self.point1().midpoint(self.point2())`.
```suggestion
    /// Get the segment's center
    #[inline(always)]
    #[doc(alias = "midpoint")]
    pub fn center(&self) -> Vec2 {
        (self.point1() + self.point2()) / 2.
    }
```
Same for 3D.

Reply:
I think Jondolf is suggesting adding the `#[doc(alias = "midpoint")]` attribute for easier discoverability of the method.

Reply:
I see but what would `self.point1().midpoint(self.point2())` be about ?

Reply:
Sorry, my comment was a bit confusing there. I just meant that this `center` method is functionally equivalent to using Glam's `midpoint` method like `segment.point1().midpoint(segment.point2())`, which is why I think adding a `midpoint` alias could be nice here.

So yeah, just adding the alias is what I intended with my suggestion.